src/Component.php

```diff
@@ -362,7 +362,11 @@ public function executeAjaxCallback(\PaymentMethod $method, &$form, &$form_state
     }
     $payment = $this->payment;
     $payment->method = $method;
-    $result = $method->controller->ajaxCallback($payment, $form, $form_state);
+    // Pass only controller specific form data to the callback.
+    $webform = new Webform($form['#node']);
+    $element = &drupal_array_get_nested_value($form['submitted'], $this->parents($webform));
+    $element = &$element['payment_method_all_forms'][$method->pmid];
```

Comment:
I would avoid the references here: The callback should not edit the element and if it does this shouldn’t have a consequence.

Reply:
Have updated the last commit.